src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -370,14 +372,42 @@ public Utf8String Substring(int index, int length)
             }
         }
 
-        // TODO: Should this be public or should focused on scenarios?
-        private int IndexOf(Utf8String value)
+        // TODO: Naive algorithm, reimplement faster
+        // TODO: Should this be public?
+        public int IndexOf(Utf8String value)
         {
-            throw new NotImplementedException();
+            if (value.Length == 0)
+            {
+                // TODO: Is this the right answer?
+                // TODO: Does this even make sense?
```

Comment:
Im not sure about that. IMO searching for "" doesn't make sense but from mathematical perspective:
https://en.wikipedia.org/wiki/Empty_set
empty set is a subset of any set so technically we should return 0.